modules/core/src/main/java/org/apache/ignite/internal/cdc/CdcManager.java

```diff
@@ -0,0 +1,104 @@
+import org.apache.ignite.IgniteSystemProperties;
+import org.apache.ignite.cdc.CdcConsumer;
+import org.apache.ignite.internal.pagemem.wal.IgniteWriteAheadLogManager;
+import org.apache.ignite.internal.pagemem.wal.record.CdcManagerRecord;
+import org.apache.ignite.internal.pagemem.wal.record.CdcManagerStopRecord;
+import org.apache.ignite.internal.processors.cache.GridCacheSharedManager;
+import org.apache.ignite.internal.processors.cache.persistence.wal.WALPointer;
+import org.apache.ignite.plugin.PluginContext;
+import org.apache.ignite.plugin.PluginProvider;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * CDC manager responsible for logic of capturing data within Ignite node. There is a contract between {@link CdcManager}
+ * and {@link CdcMain}. Communication between these components are built with WAL records. {@link CdcManager} can
+ * write some records, while {@link CdcMain} must listen to them.
+ * <ul>
+ *     <li>
+ *         This manager can write {@link CdcManagerRecord} that contains info about consumer state.
+ *         This record will be handled by {@link CdcMain} the same way it handles {@link CdcConsumer#onEvents(Iterator)}
+ *         when it returns {@code true}.
+ *     </li>
+ *     <li>
+ *         This manager must write {@link CdcManagerStopRecord} after it stopped handling records by any reason.
+ *         {@link CdcMain} will start consume records instead since {@link CdcManagerRecord#walState()} of
+ *         last consumed {@link CdcManagerRecord}.
+ *     </li>
+ *     <li>
+ *         Apache Ignite provides a default implementation - {@link CdcUtilityActiveCdcManager}. It will set the CDC mode to
+ *         {@link CdcMode#CDC_UTILITY_ACTIVE} on node start up.
+ *     </li>
+ * </ul>
+ *
+ * Ignite can be extended with custom CDC manager. It's required to implement this interface, create own {@link PluginProvider}
+ * that will return the implementation with {@link PluginProvider#createComponent(PluginContext, Class)} method.
+ * The {@code Class} parameter in the method is {@code CdcManager} class.
+ *
+ * @see CdcConsumer#onEvents(Iterator)
+ * @see CdcConsumerState#saveCdcMode(CdcMode)
+ */
+public interface CdcManager extends GridCacheSharedManager {
+    /**
+     * Callback is invoked once, after Ignite restores memory on start-up. It runs within Ignite start thread before
+     * {@link IgniteWriteAheadLogManager} enables writing new WAL records. WALPointer {@code restoredPtr} points
+     * to the head of the WAL, or {@code null} if no WAL records had written yet.
+     *
+     * This method invoked before first call of {@link #collect(ByteBuffer)}. The first collected buffer starts from
+     * the next pointer after {@code restoredPtr}.
+     *
+     * @param restoredPtr Pointer to WAL head record, {@code null} if no WAL records had written yet.
+     */
+    public void afterMemoryRestore(@Nullable WALPointer restoredPtr) throws IgniteCheckedException;
```

Comment:
Do we really nee this callback? 
Looks like we have the same pointer on the first invokation of `collect`.

Reply:
If this callback really required - please, add comment - what expected checks and behaviour of implementations.